simple/unexpected_msg.c

```diff
@@ -225,6 +239,9 @@ int main(int argc, char **argv)
 		case 'm':
 			opts.transfer_size = strtoul(optarg, NULL, 0);
 			break;
+		case 'd':
+			send_data = true;
```

Comment:
I missed this in earlier reviews.  When requesting CQ data, we should set hints->domain_attr->cq_data_size = 4 or 8 (4 is preferred), so that the provider knows to enable CQ data support or fail the fi_getinfo request.  I'm okay with this coming as a follow up path.

We should also or in FI_RX_CQ_DATA into the hints->mode, if the test can support that mode.  Quickly looking at the test, I think it can.